contracts/test/Spender.t.sol

```diff
@@ -51,8 +63,14 @@ contract SpenderTest is Test {
         // Mint 10k tokens to user
         lon.mint(user, 10000 * 1e18);
         // User approve AllowanceTarget
-        vm.prank(user);
+        vm.startPrank(user);
         lon.approve(address(allowanceTarget), type(uint256).max);
+        // Set user's mock tokens balance and approve
+        for (uint256 j = 0; j < tokens.length; j++) {
+            setERC20Balance(address(tokens[j]), user, 100);
```

Comment:
Could we use `deal` here?

And if this is possible, maybe `StrategySharedSetup.sol:122` can do the same refactoring

Reply:
`deal` would work but act like `setERC20BalanceRaw` so that's why I keep `setERC20Balance` as it can handle the `decimal` for us.

Reply:
Not sure but I think decimal is more for display purpose. Decimal is even not required in ERC20 spec. Maybe our test doesn't need to care too much about decimal.

Reply:
For RFQ/LimitOrder, it should work fine without decimal. But for AMM it probably matters unless we only test with small amount?

Reply:
I support this wrapping for decimal since conventional amount is not using `uint256` as unit but instead using weighted unit. This matters when it comes to price as @NIC619 said. For example `1 DAI approx equals to 1 USDT value` is actually `1 DAI euqals to 1e12 USDT` since DAI has decimal in 18 while USDT is in 6. I think either way we need a util to translate it.